--- loader.py	2023-06-16 10:54:58.201381900 +0000
+++ loader-new.py	2023-06-16 11:32:50.185381900 +0000
@@ -44,8 +44,11 @@
             cursor.execute(insert_query, formatted_data)
             connection.commit()
 
-            # Delete records beyond 48 hours from the SQLite table
-            delete_query = f"DELETE FROM {schema['table']} WHERE datetime(timestamp) < datetime('now', '-48 hours')"
+            # The maximum timestamp is the timestamp of the last inserted record
+            max_timestamp = formatted_data[0]
+
+            # Delete records beyond 48 hours from the maximum timestamp in the SQLite table
+            delete_query = f"DELETE FROM {schema['table']} WHERE datetime(timestamp) < datetime('{max_timestamp}', '-48 hours')"
             cursor.execute(delete_query)
             connection.commit()
 
