--- loader.py	2023-06-14 15:08:06.360030200 +0000
+++ loader-new.py	2023-06-14 15:10:53.652030200 +0000
@@ -44,6 +44,11 @@
             cursor.execute(insert_query, formatted_data)
             connection.commit()
 
+            # Delete records beyond 48 hours from the SQLite table
+            delete_query = f"DELETE FROM {schema['table']} WHERE datetime(timestamp) < datetime('now', '-48 hours')"
+            cursor.execute(delete_query)
+            connection.commit()
+
             return cursor.lastrowid > 0
 
     except sqlite3.Error as error_msg:
